lib/webdriveragent.js

```diff
@@ -59,31 +61,48 @@ class WebDriverAgent {
 
     this.updatedWDABundleId = args.updatedWDABundleId;
 
-    this.xcodebuild = new XcodeBuild(this.xcodeVersion, this.device, {
-      platformVersion: this.platformVersion,
-      platformName: this.platformName,
-      iosSdkVersion: this.iosSdkVersion,
-      agentPath: this.agentPath,
-      bootstrapPath: this.bootstrapPath,
-      realDevice: this.isRealDevice,
-      showXcodeLog: args.showXcodeLog,
-      xcodeConfigFile: args.xcodeConfigFile,
-      xcodeOrgId: args.xcodeOrgId,
-      xcodeSigningId: args.xcodeSigningId,
-      keychainPath: args.keychainPath,
-      keychainPassword: args.keychainPassword,
-      useSimpleBuildTest: args.useSimpleBuildTest,
-      usePrebuiltWDA: args.usePrebuiltWDA,
-      updatedWDABundleId: this.updatedWDABundleId,
-      launchTimeout: args.wdaLaunchTimeout || WDA_LAUNCH_TIMEOUT,
-      wdaRemotePort: this.wdaRemotePort,
-      useXctestrunFile: this.useXctestrunFile,
-      derivedDataPath: args.derivedDataPath,
-      mjpegServerPort: this.mjpegServerPort,
-      allowProvisioningDeviceRegistration: args.allowProvisioningDeviceRegistration,
-      resultBundlePath: args.resultBundlePath,
-      resultBundleVersion: args.resultBundleVersion,
-    }, this.log);
+    this.usePreinstalledWDA = args.usePreinstalledWDA;
+    this.xctestApiClient = null;
+
+    // FIXME: maybe 'this.webDriverAgentUrl' also can ignore
+    // the xcodebuild instance itself.
+    if (this.usePreinstalledWDA) {
+      this.xcodebuild = null;
```

Comment:
do not return immediately here so that we may write something after this if/else clause in the future. (do not restrict the order)